GRADLE_ARGS ignored in recursive (-r) mode

Given a `build.gradle` with the following content:
```
buildscript {
    repositories {
        maven {
            credentials {
                username usernameProperty
            }
            url "https://repo.maven.apache.org/maven2/"
        }
    }
}
```

and the following folder structure (`build.gradle` identical):
```
.
├── build.gradle
└── subdir
    └── build.gradle
```

When running `GRADLE_ARGS="-PusernameProperty=bliblablub" cdxgen -o sbom.json`:
```
Executing gradle dependencies -q --console plain -PusernameProperty=bliblablub in /home/heubeck/w/sbom-test/test
No packages were detected. If this is a multi-project gradle application set the environment variable GRADLE_MULTI_PROJECT_MODE to true and try again.
```

But when running `GRADLE_ARGS="-PusernameProperty=bliblablub" cdxgen -o sbom.json -r`:
```

FAILURE: Build failed with an exception.

* Where:
Build file '/home/heubeck/w/sbom-test/test/build.gradle' line: 5

* What went wrong:
A problem occurred evaluating root project 'test'.
> Could not get unknown property 'usernameProperty' for Credentials [username: null] of type org.gradle.internal.credentials.DefaultPasswordCredentials_Decorated.
---
```

Tried to find it by myself, but failed after just reading through the cdxgen sources. With `-r` the GRADLE_ARGS are not respected, and the "Executing gradle..." statement is also not printed, already while processing the first build.gradle.

Without the subdir, everything is fine, also with `-r`.

Tested with cdxgen 8.4.8 as well as with 8.4.0 - same behavior.

--- a/index.js
+++ b/index.js
@@ -1167,14 +1167,23 @@ const createJavaBom = async (path, options) => {
         ["true", "1"].includes(multiProjectMode) ||
         (gradleFiles.length > 1 && !["false", "0"].includes(multiProjectMode))
       ) {
-        if (DEBUG_MODE) {
-          console.log("Executing", gradleCmd, "projects in", path);
+        let gradleProjectsArgs = ["projects", "-q", "--console", "plain"];
+        if (process.env.GRADLE_ARGS) {
+          const addArgs = process.env.GRADLE_ARGS.split(" ");
+          gradleProjectsArgs = gradleProjectsArgs.concat(addArgs);
         }
-        const result = spawnSync(
+        console.log(
+          "Executing",
           gradleCmd,
-          ["projects", "-q", "--console", "plain"],
-          { cwd: path, encoding: "utf-8", timeout: TIMEOUT_MS }
+          gradleProjectsArgs.join(" "),
+          "projects in",
+          path
         );
+        const result = spawnSync(gradleCmd, gradleProjectsArgs, {
+          cwd: path,
+          encoding: "utf-8",
+          timeout: TIMEOUT_MS
+        });
         if (result.status !== 0 || result.error) {
           if (result.stderr) {
             console.error(result.stdout, result.stderr);

--- a/utils.js
+++ b/utils.js
@@ -1214,6 +1214,7 @@ const parseGradleDep = function (
     const deps = [rootProject];
     const dependenciesList = [];
     const keys_cache = {};
+    const deps_keys_cache = {};
     let last_level = 0;
     let last_purl = `pkg:maven/${rootProjectName}@${rootProjectVersion}?type=jar`;
     const first_purl = last_purl;
@@ -1246,9 +1247,11 @@ const parseGradleDep = function (
           !rline ||
           rline.trim() === "" ||
           rline.startsWith("+--- ") ||
-          rline.startsWith("--- ")
+          rline.startsWith("\\--- ")
         ) {
           last_level = 1;
+          last_purl = first_purl;
+          stack = [last_purl];
         }
       }
       while ((match = depRegex.exec(rline))) {
@@ -1265,43 +1268,43 @@ const parseGradleDep = function (
             null
           ).toString();
           purlString = decodeURIComponent(purlString);
-
-          // Filter duplicates
-          if (!keys_cache[purlString]) {
-            keys_cache[purlString] = true;
-            if (group !== "project") {
+          keys_cache[purlString + "_" + last_purl] = true;
+          if (group !== "project") {
+            // Filter duplicates
+            if (!deps_keys_cache[purlString]) {
+              deps_keys_cache[purlString] = true;
               deps.push({
                 group,
                 name: name,
                 version: version,
                 qualifiers: { type: "jar" }
               });
-              if (!level_trees[purlString]) {
-                level_trees[purlString] = [];
-              }
-              if (level == 0 || last_purl === "") {
-                stack.push(purlString);
-              } else if (level > last_level) {
-                const cnodes = level_trees[last_purl] || [];
-                cnodes.push(purlString);
-                level_trees[last_purl] = cnodes;
-                if (stack[stack.length - 1] !== purlString) {
-                  stack.push(purlString);
-                }
-              } else {
-                for (let i = level; i <= last_level; i++) {
-                  stack.pop();
-                }
-                const last_stack =
-                  stack.length > 0 ? stack[stack.length - 1] : first_purl;
-                const cnodes = level_trees[last_stack] || [];
-                cnodes.push(purlString);
-                level_trees[last_stack] = cnodes;
-                stack.push(purlString);
-              }
-              last_level = level;
-              last_purl = purlString;
             }
+            if (!level_trees[purlString]) {
+              level_trees[purlString] = [];
+            }
+            if (level == 0 || last_purl === "") {
+              stack.push(purlString);
+            } else if (level > last_level) {
+              const cnodes = level_trees[last_purl] || [];
+              cnodes.push(purlString);
+              level_trees[last_purl] = cnodes;
+              if (stack[stack.length - 1] !== purlString) {
+                stack.push(purlString);
+              }
+            } else {
+              for (let i = level; i <= last_level; i++) {
+                stack.pop();
+              }
+              const last_stack =
+                stack.length > 0 ? stack[stack.length - 1] : first_purl;
+              const cnodes = level_trees[last_stack] || [];
+              cnodes.push(purlString);
+              level_trees[last_stack] = cnodes;
+              stack.push(purlString);
+            }
+            last_level = level;
+            last_purl = purlString;
           }
         }
       }

--- a/utils.test.js
+++ b/utils.test.js
@@ -953,6 +953,7 @@ test("parse .net cs proj", async () => {
 });
 
 test("get nget metadata", async () => {
+  jest.setTimeout(240000);
   const dep_list = await utils.getNugetMetadata([
     {
       group: "",

